crates/sui-replay/src/lib.rs

```diff
@@ -343,6 +363,42 @@ pub async fn execute_replay_command(
             // TODO: clean this up
             Some((0u64, 0u64))
         }
+        ReplayToolCommand::ProfileTransaction {
+            tx_digest,
+            show_effects,
+            diag,
+            executor_version_override,
+            protocol_version_override,
+            profile_output_filepath_override,
+        } => {
+            let tx_digest = TransactionDigest::from_str(&tx_digest)?;
+            info!("Executing tx: {}", tx_digest);
+            let sandbox_state = LocalExec::replay_with_network_config(
+                rpc_url,
+                cfg_path.map(|p| p.to_str().unwrap().to_string()),
+                tx_digest,
+                safety,
+                use_authority,
+                executor_version_override,
+                protocol_version_override,
+                profile_output_filepath_override
+                    .or(Some((*DEFAULT_PROFILE_OUTPUT_PATH.clone()).to_path_buf())),
+            )
+            .await?;
+
+            if diag {
+                println!("{:#?}", sandbox_state.pre_exec_diag);
```

Comment:
Calling out for a follow-up PR -- the replay command seems to mix log output, and `println!` output -- we should standardise (and also pick an output format that fits with the rest of the CLI)